src/index.ts

```diff
@@ -6,22 +6,21 @@ import type {
 
 // Helper method to wait for a middleware to execute before continuing
 // And to throw an error when an error happens in a middleware
-function initMiddleware(middleware) {
+function initMiddleware(middleware: typeof cors) {
    return (
       req: NextApiRequest,
       res: NextApiResponse,
       options?: CorsOptions | CorsOptionsDelegate,
-   ) => {
+   ) =>
       new Promise((resolve, reject) => {
-         middleware(options)(req, res, (result) => {
+         middleware(options)(req, res, (result: typeof cors) => {
```

Comment:
remember that it is the result of the promise, so it would be to use `Error | unknown`.

```suggestion
         middleware(options)(req, res, (result: Error | unknown) => {
```